geth/jail/jail_test.go

```diff
@@ -115,6 +128,46 @@ func (s *JailTestSuite) TestFunctionCall() {
 	require.Equal(expectedResponse, response)
 }
 
+func (s *JailTestSuite) TestJailRPCAsyncSend() {
+	require := s.Require()
+
+	defer func() {
+		if err := recover(); err != nil {
+			require.NotNil(err, "Panic occured")
+		}
+	}()
+
+	require.NotNil(s.jail)
+
+	s.StartTestNode(params.RopstenNetworkID)
+	defer s.StopTestNode()
+
+	// load Status JS and add test command to it
+	s.jail.BaseJS(baseStatusJSCode)
+	s.jail.Parse(testChatID, rxSendJS)
+
+	cell, err := s.jail.GetCell(testChatID)
+	require.NoError(err)
+	require.NotNil(cell)
+
+	var wg sync.WaitGroup
+
+	// internally (since we replaced `web3.send` with `jail.Send`)
+	// all requests to web3 are forwarded to `jail.Send`
+	for i := 0; i < 5; i++ {
+		go func() {
+			wg.Add(1)
```

Comment:
`wg.Add(1)` should be invoked before a goroutine to prevent race conditions.
In your case the scheduler is possible to execute `wg.Wait()` after the for loop before starting even the first goroutine.